bundles/mapping/layerswipe/instance.js

```diff
@@ -0,0 +1,143 @@
+
+            const addToolButtonBuilder = Oskari.requestBuilder('Toolbar.AddToolButtonRequest');
+            const buttonConf = {
+                iconCls: 'tool-layer-swipe',
+                tooltip: this.loc('toolLayerSwipe'),
+                sticky: true,
+                callback: () => this.setActive(!this.active)
+            };
+            sandbox.request(this, addToolButtonBuilder('LayerSwipe', 'basictools', buttonConf));
+        },
+
+        setActive: function (active) {
+            if (active) {
+                const layer = this.getTopmostLayer();
+                if (layer) {
+                    this.registerEventListeners(layer);
+                }
+                this.showSplitter();
+                if (this.cropSize === null) {
+                    const mapSize = this.map.getSize();
+                    this.cropSize = mapSize[0] / 2;
+                }
+            } else {
+                this.unregisterEventListeners();
+                this.hideSplitter();
+            }
+            this.active = active;
+            this.map.render();
+        },
+
+        updateSwipeLayer: function () {
+            this.unregisterEventListeners();
+            const layer = this.getTopmostLayer();
+            if (layer) {
+                this.registerEventListeners(layer);
+            }
+        },
+
+        getTopmostLayer: function () {
+            const layers = this.map.getLayers().getArray().filter(layer => layer.getVisible() && !(layer instanceof VectorLayer));
+            return layers.length !== 0 ? layers[layers.length - 1] : null;
+        },
+
+        registerEventListeners: function (layer) {
+            const prerenderKey = layer.on('prerender', (event) => {
+                const ctx = event.context;
+                if (!this.active) {
+                    ctx.restore();
+                    return;
+                }
+
+                const mapSize = this.map.getSize();
+                const tl = getRenderPixel(event, [0, 0]);
+                const tr = getRenderPixel(event, [this.cropSize, 0]);
+                const bl = getRenderPixel(event, [0, mapSize[1]]);
+                const br = getRenderPixel(event, [this.cropSize, mapSize[1]]);
+
+                ctx.save();
+                ctx.beginPath();
+                ctx.moveTo(tl[0], tl[1]);
+                ctx.lineTo(bl[0], bl[1]);
+                ctx.lineTo(br[0], br[1]);
+                ctx.lineTo(tr[0], tr[1]);
+                ctx.closePath();
+                ctx.clip();
+            });
+            this.eventListenerKeys.push(prerenderKey);
+            const postrenderKey = layer.on('postrender', (event) => {
+                event.context.restore();
+            });
+            this.eventListenerKeys.push(postrenderKey);
+        },
+
+        unregisterEventListeners: function () {
+            this.eventListenerKeys.forEach(key => unByKey(key));
```

Comment:
I'm not sure what this does but if it unregisters all listeners for pre/postrender (also registered by other bundles) it will break stuff.